### PR TITLE
Add repo metadata to the UI

### PR DESCRIPTION
**Pull request by camdencheek**
@limitedmage I was feeling adventurous :) 

cc @rrhyne 

Progresses #21636 

This PR adds repo metadata to the UI. As usual with frontend PRs, this is not my schtick, so please point out anywhere I'm not hitting best practices.

This adds stars to the top right of all results, but only if the associated repo has >0 stars (we don't currently distinguish between 0 stars and "we didn't collect stars for this repo"). 

Additionally, this adds description, archived, and fork to each of the repo results.

Figma design: https://www.figma.com/file/7M1QG0FfesEx8jPuYWfwoT/Result-metadata?node-id=0%3A2247

Some screenshots: 
Screen Shot 2021-06-17 at 19 15 50
Screen Shot 2021-06-17 at 18 53 45
Screen Shot 2021-06-17 at 18 54 09
Screen Shot 2021-06-17 at 18 54 26
Screen Shot 2021-06-17 at 19 37 05

### CODE REVIEW

**Comment by camdencheek** [comment]
Looks like I've got some typescript lints to fix and one unintended spacing change. I'll fix these tomorrow, but those should be small changes, so I'll leave this open for review.

File: client/shared/src/util/stars.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Converts the number of repo stars into a string, formatted nicely for large numbers
+ */
+export const starDisplay = (repoStars?: number): string | undefined => {
```
Comment (novoselrok) [inline client/shared/src/util/stars.ts:4]: Tweak: `starDisplay` -> `formatRepositoryStarCount`, I think we usually have the `format` prefix when stringifying an object.

File: client/web/src/components/SearchResult.tsx [hunk 2]
```diff
@@ -16,27 +20,77 @@ interface Props extends ThemeProps {
...
-            {result.type === 'commit' && result.detail && (
-                <>
-                    <span className="search-result__spacer" />
-                    <Markdown className="flex-shrink-0" dangerousInnerHTML={renderMarkdown(result.detail)} />
-                </>
-            )}
-        </div>
-    )
+    const renderTitle = (): JSX.Element => {
+        const starDisplayString = starDisplay(result.repoStars)
```
Comment (novoselrok) [inline client/web/src/components/SearchResult.tsx:24]: If you change `starDisplay` to `formatRepositoryStarCount`, I'd change this to `formattedRepositoryStartCoun`. I know I'm nit picking, I'm just not a big fan of using types in variable names 😄

File: client/shared/src/util/stars.ts (hunk 1 above)
Comment (camdencheek) [reply client/shared/src/util/stars.ts:4]: Lol. I spent like 5 minutes trying to come up with the name for that function, and for the life of me, I could not come up with the word "format" 🤦

File: client/web/src/components/SearchResult.tsx (hunk 2 above)
Comment (camdencheek) [reply client/web/src/components/SearchResult.tsx:24]: I'm _always_ happy to get naming feedback 😄

**Comment by camdencheek** [comment]
frontend people: what commands do you usually run locally for lints? Every time I make a PR that touches frontend code, I somehow manage to miss a set of lints

**Comment by novoselrok** [comment]
> frontend people: what commands do you usually run locally for lints? Every time I make a PR that touches frontend code, I somehow manage to miss a set of lints

You can run `yarn build-ts && yarn all:eslint` in the repo root, and that should catch most of the issues.

**Comment by camdencheek** [comment]
This should be good to go now 👍

**Review by rrhyne — APPROVED** [review]
Looks great!

**Review by novoselrok — APPROVED** [review]

File: client/shared/src/components/ResultContainer.tsx [hunk 3]
```diff
@@ -150,6 +160,15 @@ export const ResultContainer: React.FunctionComponent<Props> = ({
                         )}
                     </button>
                 )}
+                {matchCountLabel && isRedesignEnabled && formattedRepositoryStarCount && (
```
Comment (limitedmage) [inline client/shared/src/components/ResultContainer.tsx:163]:
```suggestion
                {matchCountLabel && formattedRepositoryStarCount && (
```
We can assume `isRedesignEnabled` is not needed anymore. If you want you can completely remove it from this class.

File: client/shared/src/util/stars.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Converts the number of repo stars into a string, formatted nicely for large numbers
+ */
+export const formatRepositoryStarCount = (repoStars?: number): string | undefined => {
+    if (repoStars !== undefined) {
+        if (repoStars > 1000) {
+            return `${Math.floor(repoStars / 1000)}k`
```
Comment (limitedmage) [inline client/shared/src/util/stars.ts:7]: Alternatively, we could use `.toFixed()` to give something more precise like `14.5k`

File: client/web/src/components/SearchResult.scss [hunk 5]
```diff
@@ -8,4 +8,27 @@
...
+    &__divider-vertical {
+        border-bottom: 1px solid var(--border-color-2);
+        width: 100%;
+        margin: 0.5rem 0;
+    }
+    &__match-type {
+        white-space: nowrap;
+    }
+    &__star {
+        fill: var(--yellow);
```
Comment (limitedmage) [inline client/web/src/components/SearchResult.scss:25]: I assume this color works well in both dark and light mode?

**Review by limitedmage — APPROVED** [review]

File: sg.config.yaml [hunk 6]
```diff
@@ -46,7 +46,7 @@ env:
 
   # Required for `frontend` and `web` commands
   SOURCEGRAPH_HTTPS_DOMAIN: sourcegraph.test
-  SOURCEGRAPH_HTTPS_PORT: 3443
+  SOURCEGRAPH_HTTPS_PORT: 443
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply sg.config.yaml:49]: Nope. Thanks 👍

File: client/web/src/components/SearchResult.tsx [hunk 7]
```diff
@@ -16,27 +20,77 @@ interface Props extends ThemeProps {
...
+                                <small>Repository match</small>
+                            </div>
+                            {result.fork && (
+                                <>
+                                    <div className="search-result__divider" />
+                                    <div>
+                                        <SourceForkIcon className="search-result__icon icon-inline flex-shrink-0 text-muted" />
+                                    </div>
+                                    <div>
+                                        <small>Fork</small>
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web/src/components/SearchResult.tsx:67]: Yeah -- I struggled with this one too. I think of "forked" as "this repo has been forked", and not "this repo was forked from another".

[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/components/SearchResult.tsx:67]: @rrhyne  any suggestions here for content?

File: client/web/src/components/SearchResultMatch.scss [hunk 8]
```diff
@@ -121,4 +121,9 @@
     }
 
     border-top: 1px solid var(--border-color);
+
+    .flex-row {
+        display: flex;
+        align-items: center;
+    }
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web/src/components/SearchResultMatch.scss:128]: TIL what BEM is 😅

[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/components/SearchResultMatch.scss:128]: FYI we're moving away from BEM towards CSS modules, but this file is still using BEM. For new code we prefer CSS modules, you can tell by the extension being `.module.scss`. https://docs.sourcegraph.com/dev/background-information/web/styling#structuring-style-sheets

File: client/shared/src/util/stars.ts (hunk 4 above)
Comment (camdencheek) [reply client/shared/src/util/stars.ts:7]: That looks much nicer -- thanks for the suggestion!

File: client/web/src/components/SearchResult.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web/src/components/SearchResult.tsx:67]: One additional point: we use `fork:` and `archived:` in our search query, so it feels like it makes sense to use the same form in our results

File: client/web/src/components/SearchResult.scss (hunk 5 above)
Comment (camdencheek) [reply client/web/src/components/SearchResult.scss:25]: There was no design for dark mode (I was just given a hex value that matched the pre-defined color), and it looked okay to me, so I just set it to that color for both. Screenshots below, cc @rrhyne 

Screen Shot 2021-06-21 at 12 52 14
Screen Shot 2021-06-21 at 12 52 22

File: client/web/src/components/SearchResultMatch.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web/src/components/SearchResultMatch.scss:128]: Thanks for the info!

File: client/web/src/components/SearchResult.scss (hunk 5 above)
Comment (rrhyne) [reply client/web/src/components/SearchResult.scss:25]: This color works well for both light and dark. Should have mentioned that.

**Pull request merged**